airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -404,7 +404,7 @@ def _check_dataflow_job_state(self, job) -> bool:
         :raise: Exception
         """
         if self._wait_until_finished is None:
-            wait_for_running = job['type'] == DataflowJobType.JOB_TYPE_STREAMING
+            wait_for_running = job.get('type', '') == DataflowJobType.JOB_TYPE_STREAMING
```

Comment:
```suggestion
            wait_for_running = job.get('type') == DataflowJobType.JOB_TYPE_STREAMING
```
Should be good as well